sysfs/net_class.go

```diff
@@ -0,0 +1,135 @@
+	Address          string `fileName:"address"`            // /sys/class/net/<iface>/address
+	Broadcast        string `fileName:"broadcast"`          // /sys/class/net/<iface>/broadcast
+	Carrier          uint64 `fileName:"carrier"`            // /sys/class/net/<iface>/carrier
+	CarrierChanges   uint64 `fileName:"carrier_changes"`    // /sys/class/net/<iface>/carrier_changes
+	CarrierUpCount   uint64 `fileName:"carrier_up_count"`   // /sys/class/net/<iface>/carrier_up_count
+	CarrierDownCount uint64 `fileName:"carrier_down_count"` // /sys/class/net/<iface>/carrier_down_count
+	DevID            uint64 `fileName:"dev_id"`             // /sys/class/net/<iface>/dev_id
+	Dormant          uint64 `fileName:"dormant"`            // /sys/class/net/<iface>/dormant
+	Duplex           string `fileName:"duplex"`             // /sys/class/net/<iface>/duplex
+	Flags            uint64 `fileName:"flags"`              // /sys/class/net/<iface>/flags
+	IfAlias          string `fileName:"ifalias"`            // /sys/class/net/<iface>/ifalias
+	IfIndex          uint64 `fileName:"ifindex"`            // /sys/class/net/<iface>/ifindex
+	IfLink           uint64 `fileName:"iflink"`             // /sys/class/net/<iface>/iflink
+	LinkMode         uint64 `fileName:"link_mode"`          // /sys/class/net/<iface>/link_mode
+	MTU              uint64 `fileName:"mtu"`                // /sys/class/net/<iface>/mtu
+	NameAssignType   uint64 `fileName:"name_assign_type"`   // /sys/class/net/<iface>/name_assign_type
+	NetDevGroup      uint64 `fileName:"netdev_group"`       // /sys/class/net/<iface>/netdev_group
+	OperState        string `fileName:"operstate"`          // /sys/class/net/<iface>/operstate
+	PhysPortID       string `fileName:"phys_port_id"`       // /sys/class/net/<iface>/phys_port_id
+	PhysPortName     string `fileName:"phys_port_name"`     // /sys/class/net/<iface>/phys_port_name
+	PhysSwitchID     string `fileName:"phys_switch_id"`     // /sys/class/net/<iface>/phys_switch_id
+	Speed            uint64 `fileName:"speed"`              // /sys/class/net/<iface>/speed
+	TxQueueLen       uint64 `fileName:"tx_queue_len"`       // /sys/class/net/<iface>/tx_queue_len
+	Type             uint64 `fileName:"type"`               // /sys/class/net/<iface>/type
+}
+
+// NetClass is collection of info for every interface in /sys/class/net. The map keys
+// are interface names.
+type NetClass map[string]NetClassInterface
+
+// NewNetClass returns info for all net interfaces read from /sys/class/net/<interface>.
+func NewNetClass() (NetClass, error) {
+	fs, err := NewFS(DefaultMountPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return fs.NewNetClass()
+}
+
+// NewNetClass returns info for all net interfaces read from /sys/class/net/<interface>.
+func (fs FS) NewNetClass() (NetClass, error) {
+	return newNetClass(fs.Path("class/net"))
+}
+
+// NewNetClass returns info for all net interfaces read from /sys/class/net/<interface>.
```

Comment:
This should start with a lowercase n to document the private function. I wonder why not just merging this fuction with the public NewNetClass function, given it's the only caller.

Reply:
absolutely, merging.